recipes/mosquitto/all/conanfile.py

```diff
@@ -100,7 +108,7 @@ def package_info(self):
             self.cpp_info.components["libmosquitto"].requires.append("openssl::openssl")
             self.cpp_info.components["libmosquitto"].defines.append("WITH_TLS")
         if self.settings.os == "Windows":
-            self.cpp_info.components["libmosquitto"].system_libs.append("ws2_32")
+            self.cpp_info.components["libmosquitto"].system_libs.extend(["ws2_32", "crypt32"])
```

Comment:
Does it always require `crypt32` even when `self.options.with_tls` is `False`?

Reply:
iam doubtful but it's been too long

Reply:
Are you sure about crypt32? I can't find any result when searching over the code base:

```
$ grep ws2_32 -r /home/conan/.conan/data/mosquitto/1.6.12/_/_/source/source_subfolder/ 
/home/conan/.conan/data/mosquitto/1.6.12/_/_/source/source_subfolder/src/CMakeLists.txt:	set (MOSQ_LIBS ${MOSQ_LIBS} ws2_32)
/home/conan/.conan/data/mosquitto/1.6.12/_/_/source/source_subfolder/lib/CMakeLists.txt:	set (LIBRARIES ${LIBRARIES} ws2_32)
$ grep crypt32 -r /home/conan/.conan/data/mosquitto/1.6.12/_/_/source/source_subfolder/ 
$
```

Maybe it's getting a ride from another dependency.

Reply:
it's most likely being pulled in from openssl

Reply:
now that I checked out the code, I did this because the `_patch` method forces it all the time and it was being pulled in since the default TLS is on.

I'll fix the patch and this